controllers/gatekeeper_controller_test.go

```diff
@@ -45,37 +47,855 @@ func TestReplicas(t *testing.T) {
+	expectManifestContainerArgument(g, managerContainer, auditManifest).NotTo(HaveKey(emitAuditEventsArg))
+	// test override
+	gatekeeper.Spec.Audit = &auditOverride
+	err = crOverrides(gatekeeper, auditFile, auditManifest)
+	g.Expect(err).ToNot(HaveOccurred())
+	expectManifestContainerArgument(g, managerContainer, auditManifest).To(HaveKeyWithValue(emitAuditEventsArg, "true"))
+}
+
+func TestAllAuditArgs(t *testing.T) {
+	g := NewWithT(t)
+	auditChunkSize := int64(10)
+	auditFromCache := operatorv1alpha1.AuditFromCacheEnabled
+	constraintViolationLimit := int64(20)
+	emitEvents := operatorv1alpha1.EmitEventsEnabled
+	logLevel := operatorv1alpha1.LogLevelDEBUG
+	auditInterval := metav1.Duration{
+		Duration: time.Hour,
+	}
+	auditOverride := operatorv1alpha1.AuditConfig{
+		AuditChunkSize:           &auditChunkSize,
+		AuditFromCache:           &auditFromCache,
+		ConstraintViolationLimit: &constraintViolationLimit,
+		EmitAuditEvents:          &emitEvents,
+		LogLevel:                 &logLevel,
+		AuditInterval:            &auditInterval,
+	}
+
+	gatekeeper := &operatorv1alpha1.Gatekeeper{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: "testns",
+		},
+	}
+	// test default
+	auditManifest, err := getManifest(auditFile)
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Expect(auditManifest).ToNot(BeNil())
+	expectManifestContainerArgument(g, managerContainer, auditManifest).NotTo(HaveKey(auditChunkSizeArg))
+	expectManifestContainerArgument(g, managerContainer, auditManifest).NotTo(HaveKey(auditFromCacheArg))
+	expectManifestContainerArgument(g, managerContainer, auditManifest).NotTo(HaveKey(constraintViolationLimitArg))
+	expectManifestContainerArgument(g, managerContainer, auditManifest).NotTo(HaveKey(logLevelArg))
+	expectManifestContainerArgument(g, managerContainer, auditManifest).NotTo(HaveKey(emitAuditEventsArg))
+	expectManifestContainerArgument(g, managerContainer, auditManifest).NotTo(HaveKey(auditIntervalArg))
+	// test nil
+	err = crOverrides(gatekeeper, auditFile, auditManifest)
+	g.Expect(err).ToNot(HaveOccurred())
+	expectManifestContainerArgument(g, managerContainer, auditManifest).NotTo(HaveKey(auditChunkSizeArg))
+	expectManifestContainerArgument(g, managerContainer, auditManifest).NotTo(HaveKey(auditFromCacheArg))
+	expectManifestContainerArgument(g, managerContainer, auditManifest).NotTo(HaveKey(constraintViolationLimitArg))
+	expectManifestContainerArgument(g, managerContainer, auditManifest).NotTo(HaveKey(logLevelArg))
+	expectManifestContainerArgument(g, managerContainer, auditManifest).NotTo(HaveKey(emitAuditEventsArg))
+	expectManifestContainerArgument(g, managerContainer, auditManifest).NotTo(HaveKey(auditIntervalArg))
+	// test override
+	gatekeeper.Spec.Audit = &auditOverride
+	err = crOverrides(gatekeeper, auditFile, auditManifest)
+	g.Expect(err).ToNot(HaveOccurred())
+	expectManifestContainerArgument(g, managerContainer, auditManifest).To(HaveKeyWithValue(auditChunkSizeArg, "10"))
```

Comment:
Any reason not to use the actual variables themselves instead of hard-coding the value again?

Reply:
Yes, the only reason is to test that the argument is `--audit-chunk-size=10` where `10` is a string literal. However the `auditChunkSize` variable is an `int64`. I also wanted to test the type conversion into the proper string value because is also part of the implementation. The most important one is the Duration where 1 Hour is converted into `"3600"` string literal.

Reply:
Ok, makes sense, thanks for explaining! In that case disregard all my comments around this.